plugin/effect/src/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Effect } from "effect";
+import * as z from "zod";
+
+function zodEffect(this: z.ZodType, data: unknown) {
+  return Effect.tryPromise({
+    try: async () => this.parseAsync(data),
+    catch(error) {
+      return error as z.ZodError; // only kinda a lie...
```

Comment:
Maybe errors could be encapsulated to add support for catchTags ?

Reply:
I'd never heard of catchTags before, but it seems like a better solution would be to add `_tag: "ZodError"` to the `ZodError ` prototype.

Reply:
yes!